docs/pages/dynamic_routing.md

```diff
@@ -1,83 +1,64 @@
 ```python exec
 import reflex as rx
 from pcweb import constants, styles
+```
+
+
+# Dynamic Routes
+
+Dynamic routes in Reflex allow you to handle varying URL structures, enabling you to create flexible
+and adaptable web applications. This section covers regular dynamic routes, catch-all routes,
+and optional catch-all routes, each with detailed examples.
+
+## Regular Dynamic Routes
+
+Regular dynamic routes in Reflex allow you to match specific segments in a URL dynamically.
+
+Example:
+
+```python
 
-dynamic_routes = (
-"""
 class State(rx.State):
     @rx.var
     def post_id(self) -> str:
+        # Retrieves the value of the 'pid' parameter from the current route
+        # If 'pid' is not found, it defaults to 'no pid'
         return self.router.page.params.get('pid', 'no pid')
```

Comment:
is there a reason why we get the arg this way instead of via `self.pid`?

all dynamic route args should be accessible via constructs like `rx.State.arg_name` and `self.arg_name`